src/kv/committable_tx.h

```diff
@@ -320,6 +325,11 @@ namespace kv
     {
       root_at_read_version = r;
     }
+
+    void set_store_flags(uint8_t flags)
+    {
+      store->set_flags(flags);
+    }
```

Comment:
This calls makes the setting of this flag non-transactional. If the transaction is aborted before it commits, the flag will have been set on the store but the transaction will be not have been recorded in the global order. 

Instead, a flag should be set on this transaction and be propagated to the store when it commits (see call to `apply_changes()`).

Reply:
Hmm, I assumed the store would be rolled back properly if the transaction fails. In  that case, where does the rollback logic go? Is that even necessary when the flag is in the transaction object?

Reply:
If the transaction fails, the store is not rolled back (the rollback mechanism is used on a different code path, e.g. when a node becomes candidate/follower). For example, a transaction could fail if has conflicted with another one, or if the endpoint of this transaction throws an error. 

The only way to get this working transactionally is to set the flag on the store when we're sure that the transaction has been committed, i.e. "around" `apply_changes`.

Reply:
Ah! Got it, we need flags on both.